core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -156,6 +158,13 @@ class AdminManager(val config: KafkaConfig,
         CreatePartitionsMetadata(topic.name, assignments, ApiError.NONE)
       } catch {
         // Log client errors at a lower level than unexpected exceptions
+        case e: TopicExistsException =>
+          // We checked whether the topic exists in the controller's metadata cache above, so receiving an error that the
+          // topic exists is unexpected. However, it may be the case that the topic is currently undergoing deletion and
+          // has been removed from the cache but not fully garbage collected, in which case we'll instruct the client to retry.
+          info(s"Error processing create topic request $topic", e)
+          CreatePartitionsMetadata(topic.name, Map(), ApiError.fromThrowable(new UnknownTopicOrPartitionException(
```

Comment:
It makes sense to have the client retry in this case. It's less clear what the error code should be. It's probably surprising to see `UnknownTopicOrPartitionException` when creating a topic. I assume there was no retriable error code that would work better?

Reply:
I agree that the error is undesirable. Unfortunately there isn't an error code for generic transient errors, and all the existing errors that derive from RetriableException would be misleading in their own way. An additional error could be added, but this has the downside that older clients will interpret the error as an UnknownServerException, but perhaps that's acceptable since we're already giving them an intransient TopicExistsException. I'm open for either case, let me know what's preferred.